packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1467,52 +1468,94 @@ export default abstract class SqlIntegration
 
     return { tableData: results.rows };
   }
-  getSchemaFormatConfig(schemaFormat: SchemaFormat): SchemaFormatConfig {
+  getSchemaFormatConfig(schemaFormat: AutoMetricSchemas): SchemaFormatConfig {
     switch (schemaFormat) {
+      case "amplitude": {
+        return {
+          trackedEventTableName: `EVENTS_${
+            this.settings.schemaOptions?.projectId || `*`
+          }`,
+          eventColumn: "event_type",
+          timestampColumn: "event_time",
+          userIdColumn: "user_id",
+          anonymousIdColumn: "amplitude_id",
+          getMetricTableName: () =>
+            this.generateTablePath(
+              `EVENTS_${this.settings.schemaOptions?.projectId || `*`}`
+            ),
+          getDateLimitClause: (start: Date, end: Date) =>
+            `event_time BETWEEN '${formatDate(
+              start,
+              "yyyy-MM-dd"
+            )}' AND'${formatDate(end, "yyyy-MM-dd")}'`,
+          getAdditionalEvents: () => [],
+          getMetricWhereClause: (eventName: string) =>
+            `WHERE event_name = '${eventName}'`,
+        };
+      }
       case "ga4": {
         return {
           trackedEventTableName: "events_*",
           eventColumn: "event_name",
           timestampColumn: "TIMESTAMP_MICROS(event_timestamp)",
           userIdColumn: "user_id",
           anonymousIdColumn: "user_pseudo_id",
-          includesPagesTable: false,
-          includesScreensTable: false,
-          groupByColumns: ["event"],
-          eventHasUniqueTable: false,
-          dateFormat: "yyyyMMdd",
+          getMetricTableName: () => this.generateTablePath("events_*"),
+          getDateLimitClause: (start: Date, end: Date) =>
+            `_TABLE_SUFFIX BETWEEN '${formatDate(
+              start,
+              "yyyyMMdd"
+            )}' AND'${formatDate(end, "yyyyMMdd")}'`,
+          getAdditionalEvents: () => [],
+          getMetricWhereClause: (eventName: string) =>
+            `WHERE event_name = '${eventName}'`,
         };
       }
-      // Segment & Rudderstack
-      default: {
+      case "rudderstack":
+      case "segment":
         return {
           trackedEventTableName: "tracks",
           eventColumn: "event",
           timestampColumn: "received_at",
           userIdColumn: "user_id",
           anonymousIdColumn: "anonymous_id",
           displayNameColumn: "event_text",
-          includesPagesTable: true,
-          includesScreensTable: true,
-          groupByColumns: ["event", "received_at", "displayName"],
-          eventHasUniqueTable: true,
-          dateFormat: "yyyy-MM-dd",
+          getMetricTableName: (eventName: string) =>
+            this.generateTablePath(eventName),
+          getDateLimitClause: (start: Date, end: Date) =>
```

Comment:
This should not return anything.  We should always add a WHERE clause for the `timestampColumn`.  This function should just be for additional filtering, which only GA4 requires